### ISSUE TITLE
Online/Offline status is inaccurate

### ISSUE THREAD

**Issue by RuudschMaHinda** [nextcloud/jsxc.nextcloud#30]
I have not yet found a pattern. But I noticed some inconsistencies about users being shown as online and offline.

This only happens if someone registered as 'online' when i was online. And then they are not shown as offline even when they are logged out.

People who sent me messages while I was offline do not register as 'online' when they are logged off and I came online after they logged off.

**Comment by sualko** [comment]
cc @LEDfan

**Comment by sualko** [comment]
Related: https://help.nextcloud.com/t/nextcloud-12-and-chat-ojsx-integration-doesnt-work/12973

**Comment by LEDfan** [comment]
Hi @RuudschMaHinda this should be fixed in #32, I tested multiple scenario's. 

It would be cool if you can test if it works for you, you only need to change these lines:
 - https://github.com/nextcloud/jsxc.nextcloud/pull/32/files#diff-eb43f2572b51d3f4225004a89f02c43dR196 and https://github.com/nextcloud/jsxc.nextcloud/pull/32/files#diff-eb43f2572b51d3f4225004a89f02c43dR197
 - https://github.com/nextcloud/jsxc.nextcloud/pull/32/files#diff-eb43f2572b51d3f4225004a89f02c43dR110 and https://github.com/nextcloud/jsxc.nextcloud/pull/32/files#diff-eb43f2572b51d3f4225004a89f02c43dR111

**Comment by RuudschMaHinda** [comment]
I applied the diffs an now shortly after logging in I get the error message "No Connection. Login again."

**Comment by RuudschMaHinda** [comment]
There still is one issue:

Users that were online while I logged out, are still shown as online when they are not, when I log back in. Status changes seem to only be being checked while I am logged in.

**Comment by LEDfan** [comment]
Thanks @RuudschMaHinda for testing.

I fixed a bug and I think your issue was caused by this too. See: https://github.com/nextcloud/jsxc.nextcloud/pull/32/commits/5a8a5d06bd8fd6691ecd983ab9fe8ad2a4014c84

It would be great if you can test it too.

**Comment by RuudschMaHinda** [comment]
I moved the lines as you indicated. Restarted nginx, sql, php.

It shows newly logged in users (not shown as online before, as it should be) as online and removes them from the list of online after they logged out. **(as intended)**

It still shows a previously online user as online, even though they aren't logged in anywhere. (See last comment) -- Logging in as this user and logging back out does not remove them from being listed as online. Until they have preformed one action in the webinterface: then logging out removes them from the list of online users.

The newly logged in user does not see already online users until they preform an action in their web session. **(I'd consider this a minor inconvenience)**.

So in short: Updates to the list of users being online or offline only seem to happen until the user in the list preforms an action in the web interface. This also means, that these users still show as online even though they aren't logged in as in the comment before, until they perform an action on their web session while both users are logged in. It seems though: once every one who was shown to have the wrong status has logged in and logged off this fix seems to do the trick.

I'd say, we keep this open for observation for three days, so I can test some more strange user behaviour. And then I will either close it, or we will continue this wonderful journey.

**Comment by RuudschMaHinda** [comment]
For some reason I have not found yet a pattern.

Sometimes already logged in users do not show up as online for another user logging in until the already logged in user switches their app.

**Comment by RuudschMaHinda** [comment]
This behaviour occurs when restarting the browser (Chromium Version 58.0.3029.110 Built on Ubuntu , running on Ubuntu 16.04 (64-bit) AND Firefox 54.0 (64-bit) ).

Once the browser is loaded, even with new tabs, logging in and out, etc., everything works as intended as long as the already online users have at least switched the app. (Calendar to Files, or any other from to ). As soon as the browsers were completely closed, already logged in users only get shown as online once they did that.

**Comment by LEDfan** [comment]
Hi. I have been trying to reproduce the renaming issues you got but I don't seem able to find any bug....

However it's important to know that it can take up to 15 (maybe 25) seconds before a user goes offline when he closes his browser/tab. Maybe this is the behavior you are seeing when you say `as long as the already online users have at least switched the app`?

**Comment by LEDfan** [comment]
I just merged #32. If you encounter a new problem with the presence please reopen or open a new issue.
